tests/jballerina-unit-test/src/test/resources/test-src/jvm/type-test-expr.bal

```diff
@@ -170,6 +170,51 @@ function testIsLikeForTupleWithOutRestDescriptor() {
     assertEquals(k4 is [int, string], true);
 }
 
+type R1 readonly & record {|int i;|}?;
+type R2 (readonly & record {|int i;|})?;
+type R record {|int i;|};
+
+function testR1(R1 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR2(R2 r) returns boolean {
+    return r is record {|int i;|};
+}
+
+function testR3(R1 r) {
+    if r is R {
+        R _ = r; // no error
+    } else {
+        assertTrue(false);
```

Comment:
We can directly panic with a more specific error instead.

Reply:
Added